include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -2165,6 +2180,77 @@ namespace VectorTools
 
   namespace
   {
+    // keep first argument is non-reference since we use it
+    // with 1e-8 * number
+    template <typename number1, typename number2>
+    bool real_part_bigger_than(const number1 a,
+                               const number2 &b)
+    {
+      return a > b;
+    }
+
+    template <typename number1, typename number2>
+    bool real_part_bigger_than(const std::complex<number1> a,
+                               const std::complex<number2> &b)
+    {
+      Assert(std::abs(a.imag()) < 1e-15 , ExcInternalError());
+      Assert(std::abs(b.imag()) < 1e-15 , ExcInternalError());
```

Comment:
such tests are usually quite brittle because they assume that numbers are O(1). But if you do quantum mechanics in SI units, 1e-15 is actually quite large.

A better way of writing this is probably
```
  std::abs(a.imag) <= 1e-15*std::abs(a)
```
This makes it relative to the size of the number. Furthermore, you need `operator<=` so that you can deal with the case where a==0.